### PR TITLE
Tests: The utility to help skip tests on unsupported devices was added

### PR DESCRIPTION
**Pull request by pharret31**

### CODE REVIEW

File: packages/devextreme/testing/tests/DevExpress.animation/position.tests.js [hunk 1]
```diff
@@ -993,8 +997,11 @@ const testCollision = (name, fixtureName, params, expectedHorzDist, expectedVert
     });
 
     QUnit.test('position should return window.width() if window.outerWidth == window.innerWidth (T939748)', function(assert) {
-        const isPhone = devices.real().deviceType === 'phone';
-        if(isPhone || browser.safari) {
+        if(isRealDeviceTypeNotIn(['desktop', 'tablet'], assert)) {
```
Comment (Ambrozy) [inline packages/devextreme/testing/tests/DevExpress.animation/position.tests.js]: are we sure that new types will not appear? also it looks unreadable, I would add a separate function `isRealDeviceTypeIn`

File: packages/devextreme/testing/helpers/device.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import devices from '__internal/core/m_devices';
+
+export const isRealDeviceTypeNotIn = (deviceTypes, assert = null, message = null) => {
+    const allowedDeviceTypes = Array.isArray(deviceTypes) ? deviceTypes : [deviceTypes];
+    if(!allowedDeviceTypes.includes(devices.real().deviceType)) {
+        assert && assert.ok(true, message || 'Test should be skipped on devices not in list:  ' + deviceTypes.toString());
```
Comment (ksercs) [inline packages/devextreme/testing/helpers/device.js]: It's strange for me to see that we assert here. Your function is named as a predicate => it should basically just return true/false. I believe asserts should be nowhere besides test itself, just for readability and awareness of this special "if" corner cases

File: packages/devextreme/testing/tests/DevExpress.animation/position.tests.js (hunk 1 above)
Comment (ksercs) [reply packages/devextreme/testing/tests/DevExpress.animation/position.tests.js]: Agree with Ambrozy. Actually the most readable option to me is to have `isDesktop`, `isPhone` and `isTablet` utils, with no unification

Comment (pharret31) [reply packages/devextreme/testing/tests/DevExpress.animation/position.tests.js]: New types could appear, that's why I thought it would be better to explicitly specify which devices this test should work on

File: packages/devextreme/testing/helpers/device.js (hunk 2 above)
Comment (pharret31) [reply packages/devextreme/testing/helpers/device.js]: This test utility is for skipping test cases and it preventing code duplications, so in all the scenarios where this utility should be used it has to do an assert with a message, that's why it was extracted to a specific function

Comment (pharret31) [reply packages/devextreme/testing/helpers/device.js]: Renamed to `shouldSkipTestForDevice`

File: packages/devextreme/testing/helpers/device.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import devices from '__internal/core/m_devices';
+
+export const shouldSkipTestForDevice = (allowedDeviceTypes, assert = null, message = null) => {
```
Comment (ksercs) [inline packages/devextreme/testing/helpers/device.js]: `shouldSkipTestForDevice(['desktop'])` = "should I skip test for desktop", but you assert when it's not desktop ¯\_(ツ)_/¯

Comment (pharret31) [reply packages/devextreme/testing/helpers/device.js]: the variable name is allowedDeviceTypes, but function name means skip for a current device

Comment (ksercs) [reply packages/devextreme/testing/helpers/device.js]: >variable name is allowedDeviceTypes

Yeah, function body is well readable.
However, it's call (usage in test) is ambiguous

Comment (pharret31) [reply packages/devextreme/testing/helpers/device.js]: have put all passing params to an object to avoid ambiguity

File: packages/devextreme/testing/helpers/device.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+} = {}) => {
+    if(!Array.isArray(allowedDevices)) {
+        throw new Error('allowedDevices should be an array');
+    }
+    if(!allowedDevices.includes(devices.real().deviceType)) {
+        assert && assert.ok(true, message || 'Test should be skipped on devices not in list:  ' + allowedDevices.toString());
+        return true;
+    }
+    return false;
+};
```
Comment (Ambrozy) [inline packages/devextreme/testing/helpers/device.js:16]: The purpose of this function is to skip or pass test according to device type.
Better to not use negatives and say everything directly.
We can use `QUnit.test.if` https://qunitjs.com/api/QUnit/test.if/
```
QUnit.test.if('render', isDesktop() || isTablet(), function (assert) {
  assert.strictEqual(MyApp.render(), '<p>Hello world!</p>');
});
```

File: packages/devextreme/testing/tests/DevExpress.animation/position.tests.js [hunk 5]
```diff
@@ -964,8 +965,11 @@ const testCollision = (name, fixtureName, params, expectedHorzDist, expectedVert
     });
 
     QUnit.test('position should return window.height() if window.outerHeight == window.innerHeight (T939748)', function(assert) {
-        const isPhone = devices.real().deviceType === 'phone';
-        if(isPhone || browser.safari) {
+        if(shouldSkipOnDevice({ allowedDevices: ['desktop', 'tablet'], assert })) {
```
Comment (Ambrozy) [inline packages/devextreme/testing/tests/DevExpress.animation/position.tests.js]: isDevice(['desktop'], assert)
isNotDevice(['isPhone'])

File: packages/devextreme/testing/helpers/device.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import devices from '__internal/core/m_devices';
+
+export const shouldSkipOnDevice = ({
```
Comment (ksercs) [inline packages/devextreme/testing/helpers/device.js]: let's use plural to emphasize array should be passed
```suggestion
export const shouldSkipOnDevices = ({
```

File: packages/devextreme/testing/helpers/device.js (hunk 4 above)
Comment (ksercs) [reply packages/devextreme/testing/helpers/device.js:16]: I also think it's better to use negatives because I don't want to remember all existing devices when I want to skip just on phone. I think signature gonna be "{  devices, assert, message } "

Comment (ksercs) [reply packages/devextreme/testing/helpers/device.js:16]: regarding qunit.test.if - not sure, let's not do it now at least because there will be conflicts with existing wraps like testInActiveWindow

**Review by ksercs — APPROVED** [review]

**Review by Ambrozy — APPROVED** [review]
